lib/mapper.js

```diff
@@ -15,45 +14,40 @@ var internals = {
      * @param out the object where mapped data is written
      */
     map: function map(name, spec, eventData, out) {
-        var columns, values, i, len, column, value;
+        let columns, i, len, column, value, line;
+
+        line = {
+            event: name,
+            fields: [],
+            timestamp: null
+        };
 
         columns = Object.keys(spec);
-        values = [];
 
         for (i = 0, len = columns.length; i < len; i++) {
             column = columns[i];
             value = spec[column];
             value = (typeof value === 'function') ? value(eventData) : eventData[value];
 
-            switch (typeof value) {
-                case 'string':
-                case 'boolean':
-                case 'number':
-                    // ignore
-                    break;
-
-                case 'object':
-                    if (Array.isArray(value)) {
-                        value = String(value);
-                    } else if (value instanceof Date) {
-                        value = String(value);
-                    } else {
-                        value = Stringify(value);
+            if (column === 'time') {
+                line.timestamp = value;
+            } else {
+                if(typeof value === 'object') {
+                  Object.keys(value).forEach(function(k, v) {
```

Comment:
`v` is not a value, it's the index of the `k` in the keys array. If you want the value for that key (`k`) in the `values` object you need to look it up explicitly. (See the 3rd sentence in my original comment.)